insights/parsers/pcs_status.py

```diff
@@ -75,94 +74,114 @@
+                "pacemaker: active/enabled",
+                "pcsd: active/enabled"
+            ],
+            "Nodes configured": "3",
+            "Online": "[ myhost15 myhost16 myhost17 ]",
+            "Cluster name": "openstack",
+            "Stack": "corosync"
+        }
+
     Attributes:
         nodes (list): A list containing all node names according "PCSD Status" section
-        data (dict): A dict containing key, value for each line of the output in
-            the form::
-
-             {
-                "Resources configured": "143",
-                "PCSD Status": [
-                    "myhost15: Online",
-                    "myhost17: Online",
-                    "myhost16: Online"
-                ],
-                "Current DC": "myhost15 (1) - partition with quorum",
-                "Full list of resources": [
-                    "stonith-ipmilan-10.24.221.172\t(stonith:fence_ipmilan):\tStarted myhost15",
-                    "stonith-ipmilan-10.24.221.171\t(stonith:fence_ipmilan):\tStarted myhost16",
-                    "stonith-ipmilan-10.24.221.173\t(stonith:fence_ipmilan):\tStarted myhost15",
-                ],
-                "Daemon Status": [
-                    "corosync: active/enabled",
-                    "pacemaker: active/enabled",
-                    "pcsd: active/enabled"
-                ],
-                "Nodes configured": "3",
-                "Online": "[ myhost15 myhost16 myhost17 ]",
-                "Cluster name": "openstack",
-                "Stack": "corosync"
-             }
     """
 
     def parse_content(self, content):
         self.nodes = []
-        self.data = {}
         self.bad_nodes = []
         # according this file  https://github.com/ClusterLabs/pacemaker/blob/d078ca4a9ac72fc96073a215da2eed48939536f5/tools/crm_mon.c
         key_oneline = (
             "Cluster name", "Stack", "Current DC", "Online", "RemoteOnline", "OFFLINE", "RemoteOFFLINE", "GuestOnline")
-        key_nextline = ("Full list of resources", "Failed Actions", "PCSD Status", "Daemon Status")
+
+        # pacemaker changes some prefix after version 1.0 and 2.0
+        key_nextline_titles = {
+            "Full list of resources": ["Full list of resources", "Full List of Resources"],
+            "Failed Actions": ["Failed Actions", "Failed Resource Actions"],  # "Failed Actions" was renamed to "Failed Resource Actions" at https://github.com/ClusterLabs/pacemaker/pull/1521
+            "Failed Fencing Actions": ["Failed Fencing Actions"],
+            "PCSD Status": ["PCSD Status"],
+            "Daemon Status": ["Daemon Status"],
+            "Migration Summary": ["Migration Summary"],
+            "Fencing History": ["Fencing History"],
+            "Node List": ["Node List"],
+            "Node Attributes": ["Node Attributes"],
+            "Tickets": ["Tickets"]
+        }
+        key_nextline_values = tuple(itertools.chain(*key_nextline_titles.values()))
```

Comment:
Instead of importing the 3rd party like library `itertools`, how about just using the pythonic method?:
```
key_nextline_values = [i for ii in key_nextline_titles.values() for i in ii]
```

Reply:
After reviewing the following code, it seems to me that the `key_nextline_titles` makes things a bit complex.

How about keeping the `key_nextline` as a list, and introducing an `alias` dict as:
```python
key_nextline = [
            "Full list of resources",
            "Full List of resources",
            "Failed Resource Actions",
            "Failed Actions",
            "Failed Fencing Actions",
            "PCSD Status",
            "Daemon Status",
            "Migration Summary",
            "Fencing History",
            "Node List",
            "Node Attributes",
            "Tickets",
]
 
key_nextline_alias = {
            "Full List of resources": "Full list of resources",
            "Failed Resource Actions": "Failed Actions",  # "Failed Actions" was renamed to "Failed Resource Actions" at https://github.com/ClusterLabs/pacemaker/pull/1521
        }
 ```
 
Then, only the following changes are required in L156-157:
```
attr_name = line.split(":")[0]
self[key_nextline_alias.get(attr_name, attr_name)] = multiple_lines
```